types/async-lock/index.d.ts

```diff
@@ -15,19 +15,25 @@ declare namespace AsyncLock {
          *
          * @default 0 (Never)
          */
-        timeout?: number | undefined;
+        timeout?: number;
```

Comment:
Is there a reason why you removed the `| undefined` everywhere? This can actually be useful for users that have `exactOptionalPropertyTypes` compiler option activated.

Reply:
These types have optional parameters, so I think there is no need to use union type and `undefined`.
If these are to be assignable to `undefined, I will revert this change.

I did not know `exactOptionalPropertyTypes` compiler option. Good to know. Thank you.

Reply:
I'd leave the `| undefined` for the users using `exactOptionalPropertyTypes`. And it doesn't do any harm for all the others.